fs-index/src/index.rs

```diff
@@ -547,9 +564,16 @@ impl<Id: ResourceId> ResourceIndex<Id> {
                 .or_default()
                 .insert(path.to_path_buf());
 
+            let timpestamped_path = Timestamped {
+                item: path.to_path_buf(),
+                last_modified,
+            };
+            result
+                .added
+                .insert(id, HashSet::from([timpestamped_path]));
```

Comment:
We could wrap this into similar condition, checking that the amount of paths mapped to the id is zero. And log that duplicate was introduced in the corresponding `else` branch.

This should allow us relax this requirement to `update_one`:
```
    /// - In case of a addition, the resource was not already in the index
```

Reply:
This one’s actually different since `IndexUpdate::added` is of type `HashMap<Id, HashSet<Timestamped<PathBuf>>>`, so we need to specify the path of the added resources.